programs/ssl/ssl_base64_dump.c

```diff
@@ -205,22 +226,43 @@ size_t read_next_b64_code( char *b64, const size_t max_len )
 int main( int argc, char *argv[] )
 {
     enum { B64BUF_LEN = 4 * 1024 };
-    char b64[ B64BUF_LEN ];
+    enum { SSLBUF_LEN = B64BUF_LEN * 3 / 4 + 1 };
+
+    unsigned char b64[ B64BUF_LEN ];
+    unsigned char ssl[ SSLBUF_LEN ];
     size_t b64_counter = 0;
 
     parse_arguments( argc, argv );
 
     while( NULL != b64_file )
     {
-        size_t len = read_next_b64_code( b64, B64BUF_LEN );
-        if( len > 0)
+        size_t ssl_len;
+        size_t b64_len = read_next_b64_code( b64, B64BUF_LEN );
+        if( b64_len > 0)
         {
+            int ret;
+
             b64_counter++;
 
             if( debug )
             {
-                printf( "%zu.\n", b64_counter );
-                print_b64( b64, len );
+                printf( "%zu. Base64 code:\n", b64_counter );
+                print_b64( b64, b64_len );
+            }
+
+            ret = mbedtls_base64_decode( ssl, SSLBUF_LEN, &ssl_len, b64, b64_len );
+            if( ret != 0)
+            {
+                char buff[100];
```

Comment:
Is 100 going to be enough? Consider using malloc to allocate the exact amount of bytes you need.

Reply:
This is a buffer only for error as a string and I do not think it wold be esy to detect how big should it be before calling `mbedtls_strerror()`. 100 should be good enough because that length is used in other parts of mbedTLS, but I will change it into 256 just in case.

Reply:
I removed `char buff[100];` here and used existing `b64_buf` instead